### ISSUE TITLE
Unable to re-produce the results mentioned in the paper

### ISSUE THREAD

**Issue by AmeenAli** [hwang1996/ACME#5]
Hello,
I am trying to train the model again and I am not getting the same results mentioned in the paper <br>
I have downloaded the <b>Recipe1M</b> dataset from the offical link.
I have downloaded the following files : <b> train.tar , valid.tar , test.tar </b> and I have extracted them in the folder data and trained the network and I am getting so low results.. <br>
Am I doing something wrong or this is the way I should train? <br>
Thanks

**Comment by AugustusHsu** [comment]
Any progress on this?

**Comment by AmeenAli** [comment]
@AugustusHsu no the code is so broken this code needs a whole new refactoring ...

**Comment by AugustusHsu** [comment]
@AmeenAli I try to use tensorflow to rewrite the whole code, but I stuck in the triplet loss.
When I only use triplet loss on the TextEmbedding and ImageEmbedding then go through the fc_sia Layer, but the model can't converge.
The number of the valid triplet is always 'Batch Size'.
I can send you my code.
Can you give me some advice?

**Comment by lironghua318** [comment]
Any progress on this?
My recall start decrease after 15 epochs 
epoch,lr,I2R,R@1,R@5,R@10,R2I,R@1,R@5,R@10
15,0.0001,3.1,0.3133,0.603,0.7097,3.1,0.333,0.609,0.7177
18,0.0001,260.2,0.010,0.037,0.059,255.1,0.0143,0.0457,0.0739

And,I use data generate from [im2recipe](https://github.com/torralba-lab/im2recipe-Pytorch/blob/master/scripts/mk_dataset.py),if you train with original lmdb from recipe1M,there maybe some imgs not found
see data_loader.py line21 to check wether the img is load or just load 'white'

line21:
def default_loader(path):
    try:
        im = Image.open(path).convert('RGB')
        return im
    except:
        return Image.new('RGB', (224, 224), 'white')

**Comment by lironghua318** [comment]
When I run test.py with release modesl，the reslut is
I2T Val medR 4.2500	Recall {1: 0.2969, 10: 0.6521, 5: 0.5491000000000001}
T2I Val medR 4.8000	Recall {1: 0.2753, 10: 0.627, 5: 0.5239}
It makes me confused

**Comment by AugustusHsu** [comment]
@lironghua318 you mean the performance of the model which the author release isn't the same as the paper?
![image](https://user-images.githubusercontent.com/22841553/75325028-0e44c680-58b3-11ea-81fd-4ff2e35c4e9c.png)

**Comment by hwang1996** [comment]
> @AmeenAli I try to use tensorflow to rewrite the whole code, but I stuck in the triplet loss.
> When I only use triplet loss on the TextEmbedding and ImageEmbedding then go through the fc_sia Layer, but the model can't converge.
> The number of the valid triplet is always 'Batch Size'.
> I can send you my code.
> Can you give me some advice?

If you apply the triplet loss and L_MA on the same layer, the model training is very unstable. Please refer to the pipeline shown in our paper.
It is correct that "The number of the valid triplet is always 'Batch Size'.". Because in hard sample mining, for each anchor sample, we select the hardest sample.

**Comment by hwang1996** [comment]
Please refer to #2. 
There are some recipe_id mismatch between `train_lmdb` and `train_keys` in [im2recipe-Pytorch](https://github.com/torralba-lab/im2recipe-Pytorch), so I generated `[partition]_split.pkl` from `train_lmdb` again.

**Comment by lironghua318** [comment]
> @lironghua318 you mean the performance of the model which the author release isn't the same as the paper?
> ![image](https://user-images.githubusercontent.com/22841553/75325028-0e44c680-58b3-11ea-81fd-4ff2e35c4e9c.png)

yes
